src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -92,8 +92,9 @@ private sealed partial class UIThreadOnly : IDisposable
 
             private readonly IWaitIndicator _waitIndicator;
 
-            public readonly ITextBuffer OutputBuffer;
-            public readonly ITextBuffer StandardInputBuffer;
+            public ITextBuffer OutputBuffer { get; }
+            public ITextBuffer HistoryBuffer { get; private set; }
```

Comment:
A few problems with the buffer:

```inputContentType = _textBufferFactory.TextContentType;```

this means the buffer is plain text. So you'll not get any colorization and other language services on the content. 

Second, if you change it to language content type the services light-up. But since you have a single buffer to which you append (if I understand correctly), it won't work. Once you append 2 snippets, each by itself a valid code snippet, the content of the buffer may not be valid code anymore. 

Third, what if one snippet is a command and the next snipped is actual code. Such snippets can't be together in the same buffer.